src/webapi/selection.rs

```diff
@@ -0,0 +1,341 @@
+        js! { @(no_return)
+            @{self}.removeAllRanges();
+        };
+        Ok(())
+    }
+
+    /// Collapses the [Selection](struct.Selection.html) to a single point. The document is not
+    /// modified. If the content is focused or editable, the caret will blink there.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/collapse)
+    pub fn collapse(&self, node: Node, offset: Option<u32>) -> Result<(), TODO> {
+        js! { @(no_return)
+            @{self}.collapse(@{node}, @{offset});
+        };
+        Ok(())
+    }
+
+    /// Collapses the [Selection](struct.Selection.html) to the start of the first range in the
+    /// selection. If the content is focused or editable, the caret will blink there.
+    ///
+    /// [(Javascript
+    /// docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/collapseToStart)
+    pub fn collapse_to_start(&self) -> Result<(), TODO> {
+        js! { @(no_return)
+            @{self}.collapseToStart();
+        };
+        Ok(())
+    }
+
+    /// Collapses the [Selection](struct.Selection.html) to the end of the last range in the
+    /// selection. If the content is focused or editable, the caret will blink there.
+    ///
+    /// [(Javascript
+    /// docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/collapseToEnd)
+    pub fn collapse_to_end(&self) -> Result<(), TODO> {
+        js! { @(no_return)
+            @{self}.collapseToEnd();
+        };
+        Ok(())
+    }
+
+    /// Moves the focus of the selection to a specified point. The anchor of the selection does not
+    /// move. The selection will be from the anchor node to the new focus regardless of direction.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/extend)
+    pub fn extend(&self, node: Node, offset: Option<u32>) -> Result<(), TODO> {
+        js! { @(no_return)
+            @{self}.extend(@{node}, @{offset});
+        };
+        Ok(())
+    }
+
+    /// Sets the selection to be a range including all or parts of the two specified
+    /// [Node](struct.Node.html)s, and any content located between them.
+    ///
+    /// [(Javascript
+    /// docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/setBaseAndExtent)
+    pub fn set_base_and_extent(&self, anchor_node: Node, anchor_offset: Option<u32>, focus_node: Node, focus_offset: Option<u32>) -> Result<(), TODO> {
+        js! { @(no_return)
+            @{self}.setBaseAndExtent(@{anchor_node}, @{anchor_offset}, @{focus_node}, @{focus_offset});
+        };
+        Ok(())
+    }
+
+    /// Adds all the children of the specified [Node](struct.Node.html) to the selection. Previous
+    /// selection is lost.
+    ///
+    /// [(Javascript
+    /// docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/select_all_children)
+    pub fn select_all_children(&self, node: Node) -> Result<(), TODO> {
```

Comment:
1. The specs on this method don't say anything about throwing, but there is a big `NOTE` from which we can see that it might actually throw? Can you check how it behaves on Chrome and Firefox?
2. Accept any `INode`.